gateway/integration_tests/src/main.rs

```diff
@@ -13,24 +20,186 @@ use tracing::info;
 struct GatewayTestOpts {
     #[clap(subcommand)]
     test: GatewayTest,
-
-    #[arg(long = "gw-type")]
-    gateway_type: LightningNodeType,
 }
 
 #[derive(Debug, Clone, Subcommand)]
 enum GatewayTest {
-    ConfigTest,
+    ConfigTest {
+        #[arg(long = "gw-type")]
+        gateway_type: LightningNodeType,
+    },
+    GatewaydMnemonic {
+        #[arg(long)]
+        old_gatewayd_path: PathBuf,
+        #[arg(long)]
+        new_gatewayd_path: PathBuf,
+        #[arg(long = "gw-type")]
+        gateway_type: LightningNodeType,
+        #[arg(long)]
+        old_gateway_cli_path: PathBuf,
+        #[arg(long)]
+        new_gateway_cli_path: PathBuf,
+        #[arg(long)]
+        old_gateway_cln_extension_path: PathBuf,
+        #[arg(long)]
+        new_gateway_cln_extension_path: PathBuf,
+    },
```

Comment:
you said this is gonna be temporary until we phase out the legacy gateway multi dir structure? can we add a todo tagging it and we'll add to the tracking issue of things to phase out?

Reply:
The test is really only needed until the next release, at which point we've essentially verified that this upgrade path works. We don't need to re-run a test with old code over and over again. Yes, I will add a comment todo tag

Reply:
Added TODO